metrics/ci_integrations/test/destination/firestore/test_utils/test_data/collection_reference_mock.dart

```diff
@@ -0,0 +1,8 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:firedart/firedart.dart';
+import 'package:mockito/mockito.dart';
+
+// ignore: must_be_immutable, avoid_implementing_value_types
+class CollectionReferenceMock extends Mock implements CollectionReference {}
```

Comment:
I believe it would be great to replace all `CollectionReference` mocks with this one. The same suggestion for `DocumentReferenceMock` and `FirestoreMock` classes.